deploy/modules/aws/tidb-cluster/templates/userdata.sh.tpl

```diff
@@ -18,6 +18,8 @@ sed -i 's/LimitNPROC=infinity/LimitNPROC=1048576/' /etc/systemd/system/docker.se
 systemctl daemon-reload
 systemctl restart docker
 
+echo 'OPTIONS="--default-ulimit nofile=1024000:1024000"' >> /etc/sysconfig/docker
```

Comment:
Will this be executed again after the node is rebooted?

Reply:
No, why is this needed?

Reply:
I guess it provides CLI options for dockerd. Shouldn't it be put before `systemctl restart docker`?

Reply:
Good catch, it is mistake when I split this change from my local branch

Reply:
btw, add some comments about why this line is needed will be better

Reply:
Is it possible to configure all our ulimits via `--default-ulimit` and remove these lines?

https://github.com/pingcap/tidb-operator/blob/b7f3ef5ca7e669044951ff835f5304233148e488/deploy/modules/aws/tidb-cluster/templates/userdata.sh.tpl#L15-L17

Reply:
What I'm trying to say is that we'd better to make this idempotent.

Reply:
@cofyc I think `nofile` is enough, I will verify locally with the modification of docker.service removed.

Reply:
@tennix Nice advice, I will check if this line has been added first.

Reply:
@cofyc It turns out that we should still configure the systemd service unit according to this part of `/etc/sysconfig/docker`:

```
# The max number of open files for the daemon itself, and all
# running containers.  The default value of 1048576 mirrors the value
# used by the systemd service unit.
DAEMON_MAXFILES=1048576
```